h/models/group.py

```diff
@@ -96,6 +96,23 @@ def slug(self):
         """A version of this group's name suitable for use in a URL."""
         return slugify.slugify(self.name)
 
+    @property
+    def type(self):
```

Comment:
@lyzadanger I've implemented this as a simple Python `@property` which means that when a service wants to fetch all open groups that match a given origin it'll have to either:

1. Fetch all groups that match the origin, open or otherwise. This will _currently_ just be open groups, in fact, since no other types of groups are scoped. But in future restricted groups will get mixed in. And then iterate over the groups in Python and filter them based on `type`. Or

2. If you want to query for only open groups in the first place, then import the definitions of the types from `models.groups` and use them in the sqlalchemy query in the service.

I've realised that there is a better way to do this. `type` can be implemented as a sqlalchemy "hybrid property" which means that services can use `type` in sqlalchemy queries even though it's a computed property that doesn't really exist in the db. This will simplify the queries and also avoid services having to use the joinable/readable/writeable type definitions.

_But_ for time's sake I want to merge this as-is for now and we can change it later.